geode-core/src/integrationTest/java/org/apache/geode/management/internal/cli/functions/DestroyJndiBindingFunctionTest.java

```diff
@@ -136,15 +137,69 @@ public void destroyDataSourceIsSuccessfulWhenNoBindingFound() throws Exception {
 
   @Test
   public void destroyDataSourceFailsWhenBindingHasInvalidType() throws Exception {
+
+    when(context.getArguments()).thenReturn(new Object[] {"jndi1", true});
     when(context.getResultSender()).thenReturn(resultSender);
-    spy(destroyJndiBindingFunction);
+    DestroyJndiBindingFunction destroyFunctionSpy = spy(destroyJndiBindingFunction);
+    DataSource jndiBinding = new InvalidDataSourceClass();
+    doReturn(jndiBinding).when(destroyFunctionSpy).lookUpDataSource(any());
 
+    destroyFunctionSpy.execute(context);
     verify(resultSender).lastResult(resultCaptor.capture());
-    destroyJndiBindingFunction.execute(context);
+
 
     CliFunctionResult result = resultCaptor.getValue();
 
     assertThat(result.isSuccessful()).isEqualTo(false);
-    assertThat(result.getMessage()).contains("Data Source jndi2 has invalid type for 'destroy data-source'. 'destroy jndi-binding' command should be used.");
+    assertThat(result.getMessage()).contains(
+        "Data Source jndi1 has invalid type for destroy data-source, destroy jndi-binding command should be used.");
+  }
+
+  class InvalidDataSourceClass implements DataSource {
```

Comment:
It seems like a mock should be used instead of adding a new class (i.e. InvalidDataSourceClass) to the test.

Reply:
The reason we elected to use a new class scoped data source class was so that in the future if which data sources are valid changes this test will still be comparing against an 'invalid' data source class. We originally tried to compare it against a Managed or XAPooled data source class but ran into some trouble with that and this solution seemed more accurate.